backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/TypeImpl.java

```diff
@@ -502,13 +519,36 @@ public FeatureDirectionKind directionOf(Feature feature) {
     /**
      * <!-- begin-user-doc -->
      * <!-- end-user-doc -->
-     * @generated
+     * @generated NOT
      */
     @Override
     public EList<Membership> inheritedMemberships(EList<Type> excluded) {
-        // TODO: implement this method
-        // Ensure that you remove @generated or mark it @generated NOT
-        return null;
+        excluded.add(this);
+        Conjugation conjugator = getOwnedConjugator();
+        List<Membership> conjugatedMemberships = List.of();
+        if (conjugator != null) {
+            Type type = conjugator.getOriginalType();
+            if (type != null) {
+                conjugatedMemberships = type.getMembership().stream().toList();
+            }
+        }
+
+        List<Membership> generalMemberships = getOwnedSpecialization().stream()
+                .map(spe -> spe.getGeneral())
+                .filter(g -> g != null && !excluded.contains(g))
+                .flatMap(gen -> gen.visibleMemberships(toNamespaceExclude(excluded), false, true).stream()).toList();
+
+        Membership[] data = Stream.concat(conjugatedMemberships.stream(), generalMemberships.stream())
+                // Also inherit protected memberships
+                .filter(rel -> rel.getVisibility() != VisibilityKind.PRIVATE)
+                .toArray(Membership[]::new);
+        return new EcoreEList.UnmodifiableEList<>(this, SysmlPackage.eINSTANCE.getType_InheritedMembership(), data.length, data);
+    }
+    
+    private EList<Namespace> toNamespaceExclude(EList<Type> excluded){
```

Comment:
Please add @generated NOT. Any modified or new method must have @generated NOT annotation.

Reply:
Done.